px-oci-mon/main.go

```
@@ -305,53 +268,35 @@ func doInstall() {
 	}
 
 	if isRestartRequired {
-		logrus.Warn("Restarting portworx service")
-		doRestart()
+		logrus.Warn("Reloading + Restarting portworx service")
+		err = ociService.Reload()
+		err = ociService.Restart()
+		if err != nil {
+			logrus.Error(err)
+			os.Exit(-1)
+		}
 	} else {
 		logrus.Info("Portworx service restart not required.")
 	}
 }
 
 func doUninstall() {
 	logrus.Info("Stopping Portworx service")
-	var b bytes.Buffer
-	err := runExternalWithOutput(&b, "/bin/sh", "-c",
-		`systemctl stop portworx`)
-	if err != nil {
-		strerr := b.String()
-		if strings.Contains(strerr, " not loaded") {
-			logrus.Info(strerr)
-		} else {
-			logrus.WithError(err).Error("Could not stop Portworx service")
-			os.Exit(-1) // NOTE: CRITICAL failure !!
-		}
+	if err := ociService.Stop(); err != nil {
+		logrus.Error(err)
+		os.Exit(-1) // NOTE: CRITICAL failure !!
```

It'd be better if the caller/outer most function handles errors rather than os.Exit(-1) sprinkled around - doUninstall() should return an error. Similar comment for all cases that exit.